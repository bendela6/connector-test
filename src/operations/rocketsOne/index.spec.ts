import { Route } from "OpenApiRouter";
import { handleRocketsOne } from ".";

describe("rocketsOne operation", () => {

  const handle = ({ rocketId, provider }: { rocketId?: string, provider?: string }) => {
    const route: Route = {
      route: "/space/rockets/{rocketId}",
      operation: {
        operationId: "rocketsOne",
        responses: {},
      },
      pathParameters: {},
      headers: {},
    };

    if (rocketId) {
      route.pathParameters.rocketId = rocketId;
    }

    if (provider) {
      route.headers["x-provider"] = provider;
    }

    return handleRocketsOne(route);
  };

  it("should return a 200 response", async () => {
    const response = await handle({
      provider: "spacex",
      rocketId: "falcon9",
    });

    expect(response).toMatchSnapshot();
  });

  it("should return 404 provider not found", async () => {
    const response = await handle({
      provider: "nasa",
      rocketId: "falcon9",
    });

    expect(response).toMatchSnapshot();
  });

  it("should return a 404 rocket not found", async () => {
    const response = await handle({
      provider: "spacex",
      rocketId: "unknown",
    });

    expect(response).toMatchSnapshot();
  });

  it("should return a 400 missing provider", async () => {
    const response = await handle({
      rocketId: "falcon9",
    });

    expect(response).toMatchSnapshot();
  });

  it("should return a 400 missing rocketId", async () => {
    const response = await handle({
      provider: "spacex",
    });

    expect(response).toMatchSnapshot();
  });
});
