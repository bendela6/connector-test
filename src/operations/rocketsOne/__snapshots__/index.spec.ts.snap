// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rocketsOne operation should return 404 provider not found 1`] = `
Object {
  "body": Object {
    "message": "Provider nasa not found",
  },
  "status": 404,
}
`;

exports[`rocketsOne operation should return a 200 response 1`] = `
Object {
  "body": Object {
    "company": "SPACEX",
    "cost_per_launch": Object {
      "amount": 50000000,
      "currency": "USD",
    },
    "country": "United States",
    "id": "2",
    "main_image": "https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg",
  },
  "status": 200,
}
`;

exports[`rocketsOne operation should return a 400 missing provider 1`] = `
Object {
  "body": Object {
    "message": "Provider header is required",
  },
  "status": 400,
}
`;

exports[`rocketsOne operation should return a 400 missing rocketId 1`] = `
Object {
  "body": Object {
    "message": "Rocket ID is required",
  },
  "status": 400,
}
`;

exports[`rocketsOne operation should return a 404 rocket not found 1`] = `
Object {
  "body": Object {
    "message": "Not Found",
  },
  "status": 404,
}
`;
